interfaces/kalosm-common/src/cache.rs

```diff
@@ -138,6 +140,20 @@ impl ModelLoadingProgress {
     }
 }
 
+#[derive(Debug, thiserror::Error)]
+pub enum CacheError {
+    #[error("Hugging Face API error: {0}")]
+    HuggingFaceApi(#[from] hf_hub::api::sync::ApiError),
+    #[error("Unable to get file metadata for {0}: {1}")]
+    UnableToGetFileMetadata(PathBuf, #[source] tokio::io::Error),
+    #[error("IO error: {0}")]
+    Io(#[from] std::io::Error),
+    #[error("HTTP error: {0}")]
+    Http(#[from] reqwest::Error),
+    #[error("Unexpected status code: {0}")]
+    UnexpectedStatusCode(StatusCode),
```

Comment:
This could be expanded out to provide a better error message for unauthorized status codes (e.g. https://github.com/floneum/floneum/issues/280)